internal/report/report.go

```diff
@@ -433,7 +433,19 @@ func PrintAssembly(w io.Writer, rpt *Report, obj plugin.ObjTool, maxFuncs int) e
 	}
 
 	if len(syms) == 0 {
-		return fmt.Errorf("no matches found for regexp: %s", o.Symbol)
+		if address == nil { // User provides a symbol instead of an address
```

Comment:
Should we add tests covering the new code paths?

Reply:
The report_test.go did not have this method PrintAssembly set up, so in order to test this code path, I think we need to mock the IO.writer and plugin.ObjTool. I was kind of hope that this PR being a simple one that just makes the error message more clear to the user. Do you think it is worth the effort?

Reply:
Ok, I guess. But I don't have a good way of knowing that this change works as intended.

Reply:
I added the test, as I realized that there is a testdata that I can make use of, without too much effort to mocking it.

Also, I think due to some heuristic logic we place in to speed up the symbols finding process, I think essentially it is impossible to show matched symbols that do not have samples (unless it is an address). I have updated the comments in the code to make it more clear.